### ISSUE TITLE
Light is too bright when exported to Android with GLES2

### ISSUE THREAD

**Issue by Toshiwoz** [godotengine/godot#34257]
**Godot version:** 
3.2 beta3

**OS/device including version:** 
Huawei Y6 (MYA-L11), android 6.0

**Issue description:**
Since beta 3 when exporting to Android with GLES2 directional light is too bright compared to when I run on windows. Not sure about other light sources, spot and omni light.

![image](https://user-images.githubusercontent.com/21297356/70584662-dbcadd80-1b8f-11ea-9667-6ed46b56605d.png)

![Screenshot_2019-12-10-20-57-57](https://user-images.githubusercontent.com/21297356/70584855-67446e80-1b90-11ea-8dd6-3ee8a30510a3.png)

Scene is the same, I didn't change anything.

**Steps to reproduce:**
See the project I made for this issue. Compare the android export (apk not included).

**Minimal reproduction project:**
Made one just for this case.
[simplemarsscene.zip](https://github.com/godotengine/godot/files/3948045/simplemarsscene.zip)

**Comment by Toshiwoz** [comment]
I made 1 more test, reconverting back to GLES3.0 and the strange thing is that now is too bright even there (PC and Android, the same).. Anyhow, I remember the previous versions where not doing that.
I'll wait till another beta or RC to see if it's only a matter of playing around with the parameters, I guess somehow they could have changed due to some GLES2 update(?).

**Comment by clayjohn** [comment]
You forgot to check "Is sRGB"

**Comment by Toshiwoz** [comment]
> 
> 
> You forgot to check "Is sRGB". That will fix your issue in GLES3.
> 
> Ill see if I can reproduce the android issue later tonight.

I'm quite sure I haven't touched that, and it was looking ok with beta 2... I'll try that anyhow.

**Comment by clayjohn** [comment]
I can reproduce the issue on GLES2/android on beta3. It appears it is an issue with vertex-lit materials. 

In GLES3 you are specifying sRGB colors and then rendering them as linear colors, so you need to set "Is sRGB" to true. When you have it properly set on all six of your materials then the sphere renders fine in GLES3, both on desktop and on mobile.

**Comment by clayjohn** [comment]
Here is what it looks like in GLES2/Android beta2 

![Screenshot_20191216-211054](https://user-images.githubusercontent.com/16521339/70966607-aa975500-2048-11ea-85e6-e61b7fa1d0aa.png)

**Comment by clayjohn** [comment]
Ive figured out the root cause. It looks like on android the main scene shader isn't properly using the radiance mip-maps. So the specular is always way too crisp. In your test scene it looks like it is simply too bright. 

I have started debugging it and it looks like mip-maps are being generated on android, they just arent being used properly.

**Comment by NeoSpark314** [comment]
@clayjohn not sure; but maybe this issue/comment is realted to the problem: https://github.com/godotengine/godot/issues/33633#issuecomment-554537340 there I noticed that on android (quest) the second mip-level was black

**Comment by Toshiwoz** [comment]
> 
> 
> Here is what it looks like in GLES2/Android beta2
> 
> ![Screenshot_20191216-211054](https://user-images.githubusercontent.com/16521339/70966607-aa975500-2048-11ea-85e6-e61b7fa1d0aa.png)

That's weird, this is how it looks on my mobile:
![image](https://user-images.githubusercontent.com/21297356/71006187-eebf3f80-20b2-11ea-89b0-066348c2d913.png)
It is still too reflective, but that's because it has specular to 0.5.

**Comment by clayjohn** [comment]
@NeoSpark314 I definitely think it is related. Im noticing the black second mip level too. Now that I realized I can use render doc on Android I'll be able to really dig into it.

**Comment by Toshiwoz** [comment]
I don't know how useful this information is but, as I was trying to put the same rendering quality on desktop as mobile I realized that changing the shading parameters made it superbright as on mobile.

This is with the default settings:
![image](https://user-images.githubusercontent.com/21297356/71481079-68011580-27ca-11ea-9515-bd350b7ac110.png)
![image](https://user-images.githubusercontent.com/21297356/71481089-7818f500-27ca-11ea-9a89-b89a249b4e74.png)

And this with same as on mobile:
![image](https://user-images.githubusercontent.com/21297356/71481146-b6161900-27ca-11ea-8b4e-ca5855a6b53b.png)
![image](https://user-images.githubusercontent.com/21297356/71481171-c3330800-27ca-11ea-8cae-6bb253093eb9.png)

**Comment by Zireael07** [comment]
Can you try fiddling with the parameters on desktop to see which one exactly makes the difference?

**Comment by Toshiwoz** [comment]
> Can you try fiddling with the parameters on desktop to see which one exactly makes the difference?

It's only when all of them are changed that you see it back to normal, uncheck only one and you'll see it too bright.
As an example, by just checking force vertex shading:
![image](https://user-images.githubusercontent.com/21297356/71481932-fc20ac00-27cd-11ea-95a9-7732bad9ceb9.png)
![image](https://user-images.githubusercontent.com/21297356/71481940-06db4100-27ce-11ea-8f26-5422adf766d3.png)

**Comment by clayjohn** [comment]
I spent a few hours on this issue before the holidays but ran out of time to fix it. I couldn't find any obvious reasons for it to happen. But indeed is connected to blinn shading and vertex shading. Unfortunately we will have to revisit after 3.2 is released.

**Comment by Toshiwoz** [comment]
That's really unfortunate. Somehow a regression that's going to stay for a final release.
Hopefully a 3.2.1 will be out soon after.. 😔

**Comment by Calinou** [comment]
@Toshiwoz You could decrease the lights' intensity when vertex shading is enabled as a workaround. Attach this script to every light:

```gdscript
func _ready():
    if ProjectSettings.get_setting("rendering/quality/shading/force_vertex_shading"):
        energy *= 0.5
```

**Comment by CarlGustavAlbertDwarfsteinYung** [comment]
I get a similar issue but on desktop Windows 10 GTX 1650 Super. It looks normal with secularity disabled. It appears to happen only for light calculations interacting with secularity. Is almost as if roughness defaults to 0 for Vertex Shading. Ambient in the distance appears to look normal if not hit by a light source.

Force Vertex Shader (ON)
![image](https://user-images.githubusercontent.com/36826319/100028495-189ddd80-2dbd-11eb-93fd-eec00e8017c2.png)
Force Vertex Shader (ON)  with specular_disabled
![image](https://user-images.githubusercontent.com/36826319/100028339-b3e28300-2dbc-11eb-8185-9629d90b06ef.png)
without Force Vertex Shader (OFF)
![image](https://user-images.githubusercontent.com/36826319/100028055-053e4280-2dbc-11eb-8679-83d50818dbb2.png)

Should I make a new issue? I cannot find a related issue other than this one.

**Comment by akien-mga** [comment]
Fixed by #51410.
